frontend/ui/widget/inputdialog.lua

```diff
@@ -723,6 +724,52 @@ function InputDialog:_addScrollButtons(nav_bar)
                 end,
             })
         end
+        -- Add a button to go to the line by its number in the file
+        if self.fullscreen then
+            table.insert(row, {
+                text = "Go",
+                callback = function()
+                    local cur_line_num, last_line_num = self._input_widget:getLineNums()
+                    input_dialog = InputDialog:new{
+                        title = _("Enter line number"),
+                        input_hint = T("%1 (1 - %2)", cur_line_num, last_line_num),
+                        input_type = "number",
+                        stop_events_propagation = true, -- avoid interactions with upper InputDialog
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(input_dialog)
+                                        if not self.keyboard_hidden then
+                                            self:onShowKeyboard()
```

Comment:
Is that needed? That's the upper widget keyboard, I think it is just covered by the new keyboard, is is not?

Reply:
IIRC, there's a super-ugly quirk involved in the text editor plugin where you can indeed get two VirtualKeyboard instance stacked on top of each other.

I don't rightly remember all the details, though :s.

Reply:
Not needed, tested.